contracts/TsteemPool.sol

```diff
@@ -0,0 +1,289 @@
+pragma solidity ^0.5.0;
+
+import "./SafeMath.sol";
+import "./Ownable.sol";
+
+interface ERC20Token {
+    function transfer(address _to, uint256 _value) external returns (bool);
+
+    function transferFrom(
+        address sender,
+        address recipient,
+        uint256 amount
+    ) external returns (bool);
+
+    function balanceOf(address account) external view returns (uint256);
+}
+
+interface PeanutsPool {
+    function withdrawPeanuts() external;
+
+    function getPendingPeanuts() external view returns (uint256);
+}
+
+contract TsteemPool is Ownable {
+    using SafeMath for uint256;
+    struct Delegator {
+        uint256 tsteemAmount; // deposted tsteem
+        uint256 availablePeanuts; // minted peanuts for user
+        uint256 debtRewards; // rewards debt
+        bool hasDeposited; // set true when first time deposit
+    }
+    mapping(address => Delegator) public delegators;
+    address[] public delegatorsList;
+    uint256 public shareAcc;
+    uint256 public totalDepositedTSTEEM;
+    address public devAddress;
+    uint256 public lastRewardBlock;
+
+    ERC20Token Pnuts;
+    ERC20Token Tsteem;
+    PeanutsPool PnutPool;
+
+    event DepositTSTEEM(address delegator, uint256 amount);
+    event WithdrawTSTEEM(address delegator, uint256 amount);
+    event WithdrawPeanuts(address delegator, uint256 amount);
+
+    modifier onlyDelegator() {
+        require(
+            delegators[msg.sender].hasDeposited,
+            "Account is not a delegator"
+        );
+        _;
+    }
+
+    constructor(
+        address _punts,
+        address _pnutPool,
+        address _tsteem
+    ) public {
+        Pnuts = ERC20Token(_punts);
+        Tsteem = ERC20Token(_tsteem);
+        PnutPool = PeanutsPool(_pnutPool);
+
+        shareAcc = 0;
+        totalDepositedTSTEEM = 0;
+        lastRewardBlock = 0;
+    }
+
+    // should do this before game start
+    function setDevAddress(address dev) public onlyOwner {
+        require(dev != address(0), "Invalid dev address");
+        devAddress = dev;
+    }
+
+    function deposit(uint256 _amount) public {
+        if (_amount == 0) return;
+
+        // totalDepositedTSTEEM == 0 means there is not delegator exist. When first delegator come,
+        // we set lastRewardBlock as current block number, then our game starts!
+        // and reward before this moment shoud be sent to dev
+        if (totalDepositedTSTEEM == 0) {
+            PnutPool.withdrawPeanuts();
+            Pnuts.transfer(devAddress, Pnuts.balanceOf(address(this)));
```

Comment:
If no user have delegated TSTEEM, why this contract holds PNUT?

Reply:
Beacuse there is no on deposit tsteem.So we transfer the balance pnut to dev address.

Reply:
After users convert the steem to tsteem, some steem are mining, even if no one stake the tsteem

Reply:
That make sense :)